modules/openapi-generator/src/main/resources/r/api.mustache

```diff
@@ -543,24 +547,21 @@
       if (local_var_resp$status_code >= 200 && local_var_resp$status_code <= 299) {
         {{#returnType}}
         {{#isPrimitiveType}}
-        local_var_content <- local_var_resp$response
-          local_var_resp, "text", encoding = "UTF-8", simplifyVector = FALSE
-        )
         # save response in a file
         if (!is.null(data_file)) {
-          write(local_var_content, data_file)
+          private$WriteFile(local_var_resp, data_file)
```

Comment:
`private$WriteFile` both fixes a bug and introduces a feature

1. The bug was that we need to use `writeBin()`, not `write()`, when handling binary data (e.g., a gzipped directory of data)
2. The feature is that when the response is coverable to text, the data will be written as a text file instead of writing the raw response vector